Homework 3

EmelyanovPavel opened the pull request:
Homework on Operators of C++ topic

Commit:
Exercises 1-6

Commit:
homework from 4-03-2025

EmelyanovPavel committed:
Update Homework on Operators of C++ topic.cpp

tak333 reviewed (changes requested):
Картинки классные!

Но весь код в комментариях, не проверить.
Разносим по функциям, как указано в правилах оформления на диске. [Дублирую](https://drive.google.com/drive/folders/1-ZnMxxcZYwEAye4OtbkmYHP1rcC395FC?hl=ru), на всякий случай.

EmelyanovPavel committed:
Update Homework on Operators of C++ topic.cpp

Коды с отступами

EmelyanovPavel committed:
Добавлены функции войд + функция инт мэйн с вызовами функций войд, подключенные библиотеки под первым комментарием, где название темы

EmelyanovPavel committed:
Update Homework on Operators of C++ topic.cpp

имена переменных и названия функций написаны с маленькой буквы

tak333 commented on Homework on Operators of C++ topic.cpp:
```diff
@@ -0,0 +1,244 @@
+    int x, y;
+    std::cin >> x >> y;
+
+    if (x >= 0 && x < 8 && y >= 0 && y < 8) //if the point is inside the shaded area, print "Yes". 
```
Какой вариант вы делали?

tak333 commented on Homework on Operators of C++ topic.cpp:
```diff
@@ -0,0 +1,244 @@
+    std::cout << "sign: ";
+    char sign = std::cin.get();
+
+    if (sign == 'c')
```
Реализуйте это через switch.

tak333 commented on Homework on Operators of C++ topic.cpp:
```diff
@@ -0,0 +1,244 @@
+
+        std::cout << "Perimeter: " << perimeter << '\n';
+
+        if (arg <= 0)
```
В каком случае данная переменная будет отрицательной?

tak333 commented on Homework on Operators of C++ topic.cpp:
```diff
@@ -0,0 +1,244 @@
+
+//Exercise 3
+//15) 
+void task3_15()
```
Данное задание реализовывается с использованием трёх видов циклов.

tak333 reviewed (changes requested):
В целом, написано хорошо!
Есть небольшие замечания.

EmelyanovPavel committed:
Update Homework on Operators of C++ topic.cpp

Добавлены в код: оператор свитч для второй функции войд; операторы фор, вайл и ду-вайл для третьей функции войд. Имена функций войд изменены.

tak333 commented on Homework on Operators of C++ topic.cpp:
```diff
@@ -5,123 +5,135 @@
     int x, y;
     std::cin >> x >> y;
 
     if (x >= 0 && x < 8 && y >= 0 && y < 8) //if the point is inside the shaded area, print "Yes". 
```
Так и не поняла, какой вы вариант сделали, под 15 другая форма.

tak333 commented on Homework on Operators of C++ topic.cpp:
```diff
@@ -5,123 +5,135 @@
-        std::cout << "radius: ";
-        int radius;
+    char figure;
+    double radius, length, width, side1, side2, side3, p, circlePerimeter, circleArea, rectanglePerimeter, rectangleArea, trianglePerimeter, triangleArea;
```
Создавайте переменные по мере необходимости: в каждом case, если нужно.
Правда в таком случае нужно ограничивать время жизни каждой переменной и добавлять фигурные скобки вокруг блока.

tak333 replied:
Предыдущее именование было говорящим и достаточным. Не стоит сильно усложнять.

tak333 commented on Homework on Operators of C++ topic.cpp:
```diff
@@ -5,123 +5,135 @@
 
-        std::cout << "radius: ";
-        int radius;
+    char figure;
```
Переменные создаём по мере необходимости.

tak333 replied:
Не исправлено.

EmelyanovPavel committed:
Update Homework on Operators of C++ topic.cpp

Добавлен рисунок к 15 примеру из первого упражнения для первой функции void и изменены переменные типа double после объявления строки во второй функции void.

tak333 commented on Homework on Operators of C++ topic.cpp:
```diff
@@ -0,0 +1,258 @@
+    int x, y;
+    std::cin >> x >> y;
+
+    if (x >= 0 && x < 8 && y >= 0 && y < 8) //if the point is inside the shaded area, print "Yes". 
```
Где работа с окружностью?

EmelyanovPavel replied:
здесь работа с графиком функции

tak333 commented on Homework on Operators of C++ topic.cpp:
```diff
@@ -0,0 +1,258 @@
+    if (x == 0 || (2 * pow(x, 5) - 1) < 0)
+        return NAN; 
+
+    return log(fabs(3 * x)) * sqrt(2 * pow(x, 5) - 1);
```
std:: для библиотечных функций

EmelyanovPavel committed:
Update Homework on Operators of C++ topic.cpp

Изменен код для 15 варианта из упражнения 1; все коды написаны по заданиям из файла пдф.

tak333 commented on Homework on Operators of C++ topic.cpp:
```diff
@@ -16,36 +16,29 @@
+    
+    double boundary = -x; //function value on a straight line
+    
+    if (y == boundary) { 
```
Вы не планируете отвечать на мои предыдущие комментарии?
У вас 15 вариант, вот график: ![image](https://github.com/user-attachments/assets/ddeec275-8696-4afb-8360-38e48b4d8ca1)
Где работа с окружностью?

tak333 reviewed (changes requested):
Не исправлены замечания.

EmelyanovPavel committed:
Update Homework on Operators of C++ topic.cpp

Оформлен рисунок в виде ссылки для первой функции войд и в её теле; объявлены дополнительные переменные в кейсах оператора свитч во второй функции войд.

tak333 reviewed (changes requested):
Не исправлены замечания.
Изучите работу с графиком окружности.

EmelyanovPavel committed:
Update Homework on Operators of C++ topic.cpp

Код в теле первой функции void написан так, как написано в учебнике на юрайте. Без использования константы eps (равна 1е-12) код работает неправильно.

tak333 commented on Homework on Operators of C++ topic.cpp:
```diff
@@ -11,36 +10,25 @@
-    fs.open(url);
-    fs.close();
-
+    constexpr auto rr = 100;
```
Не используйте auto без необходимости.

tak333 commented on Homework on Operators of C++ topic.cpp:
```diff
@@ -11,36 +10,25 @@
-    if (y == boundary) {
-
+    
+    const auto a = x * x + y * y;
```
Неочевидные имена: почему а и b?
Ниже код в условии сложно читается: попробуйте его сделать максимально понятным

EmelyanovPavel replied:
В остальных кодах ниже переменные объявлены по необходимости

EmelyanovPavel replied:
Как сделать приведенный ниже код более понятным?

tak333 replied:
Используйте именование подобно тому, что изучается в курсе математики, например.
Сделайте имена переменных более говорящими.

tak333 reviewed (changes requested):
Причешите оформление и принято.

EmelyanovPavel committed:
Update Homework on Operators of C++ topic.cpp

Написаны коды более понятно.

EmelyanovPavel committed:
Update Homework on Operators of C++ topic.cpp

Написаны говорящие имена переменных в первой, второй и третьей функциях void.

tak333 reviewed (changes requested):
Ваши имена просто дублируют код, а не помогают его читать.
Также как и iks, igrek - чем стандартные x и y не угодили?
Если вы считаете сумму / разность, то не надо называть переменную summaOfTwoVariables и пр... Это и так прекрасно видно. Надо подчеркнуть, что именно вы таким образом нашли. Тангенс угла наклона, коэффициент, свободный член, радиус в квадрате, и пр., и пр.

EmelyanovPavel committed:
Update Homework on Operators of C++ topic.cpp

Изменены имена переменных в первой функции войд.
